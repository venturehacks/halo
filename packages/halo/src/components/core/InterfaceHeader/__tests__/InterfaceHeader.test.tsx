import { render } from 'enzyme';
import React from 'react';

import { InterfaceHeader } from '../index';

describe('InterfaceHeader', () => {
  test('smoke', () => {
    const component = render(
      <InterfaceHeader
        byline="Customize your stuff!"
        title="General Settings"
        page
      />,
    );
    expect(component.text()).toContain('General Settings');
    expect(component.text()).toContain('Customize your stuff!');
  });

  test('className', () => {
    const component = render(
      <InterfaceHeader className="mb-10" title="General Settings" page />,
    );
    expect(component.hasClass('mb-10')).toBeTruthy();
  });

  describe('lacking hierarhcy', () => {
    test('throws exception ', () => {
      expect(() =>
        render(<InterfaceHeader title="This should throw an exception" />),
      ).toThrow();
    });
  });

  describe('snapshot', () => {
    test('byline', () => {
      const component = render(
        <InterfaceHeader byline="Byline" title="Title" page />,
      );

      expect(component).toMatchSnapshot();
    });

    describe('hierarchy', () => {
      test('page', () => {
        const component = render(
          <InterfaceHeader title="General Settings" page />,
        );

        expect(component).toMatchSnapshot();
      });

      test('section', () => {
        const component = render(
          <InterfaceHeader title="Unbiased Sourcing" section />,
        );

        expect(component).toMatchSnapshot();
      });

      test('subsection', () => {
        const component = render(
          <InterfaceHeader title="Diversity" subsection />,
        );

        expect(component).toMatchSnapshot();
      });

      test('mini', () => {
        const component = render(
          <InterfaceHeader title="Ideal next opportunity" mini />,
        );

        expect(component).toMatchSnapshot();
      });
    });
  });
});
