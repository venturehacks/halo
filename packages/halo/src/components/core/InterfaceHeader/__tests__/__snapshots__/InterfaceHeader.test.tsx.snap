// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterfaceHeader snapshot byline 1`] = `
<header
  class="text-dark-aaaa font-medium antialiased font-galano text-xl"
>
  <div>
    Title
  </div>
  <div
    class="text-dark-aa font-normal font-arbeit text-md mt-2"
  >
    Byline
  </div>
</header>
`;

exports[`InterfaceHeader snapshot hierarchy mini 1`] = `
<header
  class="text-dark-aaaa font-medium antialiased text-sm"
>
  Ideal next opportunity
</header>
`;

exports[`InterfaceHeader snapshot hierarchy page 1`] = `
<header
  class="text-dark-aaaa font-medium antialiased font-galano text-xl"
>
  General Settings
</header>
`;

exports[`InterfaceHeader snapshot hierarchy section 1`] = `
<header
  class="text-dark-aaaa font-medium antialiased text-lg"
>
  Unbiased Sourcing
</header>
`;

exports[`InterfaceHeader snapshot hierarchy subsection 1`] = `
<header
  class="text-dark-aaaa font-medium antialiased text-md"
>
  Diversity
</header>
`;
